### PR TITLE
fix: Await Metadata file updates to avoid race conditions

### PR DESCRIPTION
**Pull request by awsluja**
#### Description of changes
Fixed occurrences of un-awaited promises in our push logic. This will ensure better consistency when running e2e tests that should await file updates on push.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by akshbhu — APPROVED** [review]

File: packages/amplify-category-auth/src/provider-utils/awscloudformation/utils/amplify-meta-updaters.ts [hunk 1]
```diff
@@ -1,69 +1,87 @@
 import * as path from 'path';
-import { JSONUtilities, pathManager, $TSAny } from 'amplify-cli-core';
+import {
+  JSONUtilities, pathManager, $TSAny, $TSContext,
+} from 'amplify-cli-core';
 import { hostedUIProviders } from '../assets/string-maps';
 import { AuthParameters } from '../import/types';
 
+type FrontEndConfig = {
```
Comment (lazpavel) [inline packages/amplify-category-auth/src/provider-utils/awscloudformation/utils/amplify-meta-updaters.ts:8]: nit: I believe `Frontend` casing is more prevalent in our code

**Review by pavellazar — APPROVED** [review]
LGTM

**Review by edwardfoyle — COMMENTED** [review]
How many errors pop up if we add https://typescript-eslint.io/rules/no-floating-promises/ to our eslint config? If it's only a handful can we add that rule and fix them in this PR?

Otherwise, adding that lint rule and fixing any identified errors should be a fast follow

File: packages/amplify-category-auth/src/provider-utils/awscloudformation/utils/amplify-meta-updaters.ts [hunk 2]
```diff
@@ -1,69 +1,87 @@
...
-    const authParameters = JSONUtilities.readJson<AuthParameters>(parametersJSONPath)!;
-
-    if (authParameters.dependsOn) {
-      options.dependsOn = authParameters.dependsOn;
-    }
+export const getPostAddAuthMetaUpdater = (context: $TSContext,
+  resultMetadata: {
+    service: string;
+    providerName: string
+  }) => (resourceName: string): string => {
```
Comment (jhockett) [inline packages/amplify-category-auth/src/provider-utils/awscloudformation/utils/amplify-meta-updaters.ts:34]: nit: I prefer having parameters with their own indentation block for readability, but not a blocker
```suggestion
export const getPostAddAuthMetaUpdater = (
  context: $TSContext,
  resultMetadata: {
    service: string;
    providerName: string
  },
) => (resourceName: string): string => {
```

File: .circleci/config.base.yml [hunk 3]
```diff
@@ -115,6 +115,8 @@ jobs:
       - run:
           name: Collect code coverage
           command: yarn coverage
+    environment:
+      NODE_OPTIONS: --max-old-space-size=4096
```
Comment (jhockett) [inline .circleci/config.base.yml:119]: This should be a separate PR for better traceability

**Comment by awsluja** [comment]
> 

@edwardfoyle Adding the lint rule isn't working too well without full TS typings in most places so I'll backlog that for now and could pull it into a larger refactor
>
